types/compression/index.d.ts

```diff
@@ -22,49 +51,102 @@ declare namespace e {
+         *   them better. The effect is to force more Huffman coding and less string matching; it is somewhat intermediate
+         *   between `zlib.Z_DEFAULT_STRATEGY` and `zlib.Z_HUFFMAN_ONLY`.
+         * - `zlib.Z_FIXED` Use to prevent the use of dynamic Huffman codes, allowing for a simpler decoder for special applications.
+         * - `zlib.Z_HUFFMAN_ONLY` Use to force Huffman encoding only (no string match).
+         * - `zlib.Z_RLE` Use to limit match distances to one (run-length encoding). This is designed to be almost as
+         *    fast as `zlib.Z_HUFFMAN_ONLY`, but give better compression for PNG image data.
+         *
+         * **Note** in the list above, `zlib` is from `zlib = require('zlib')`.
          */
         strategy?: number;
 
         /**
-         * See https://github.com/expressjs/compression#threshold regarding the usage.
+         * The byte threshold for the response body size before compression is considered for the response, defaults to
+         * 1kb. This is a number of bytes or any string accepted by the bytes module.
+         *
+         * **Note** this is only an advisory setting; if the response size cannot be determined at the time the response
+         * headers are written, then it is assumed the response is *over* the threshold. To guarantee the response size
+         * can be determined, be sure set a `Content-Length` response header.
+         *
+         * @see {@link https://www.npmjs.com/package/bytes|`bytes` module}
+         * @see {@link https://github.com/expressjs/compression#threshold|`threshold` documentation}
          */
         threshold?: number | string;
 
         /**
-         * See https://github.com/expressjs/compression#windowbits regarding the usage.
+         * The default value is `zlib.Z_DEFAULT_WINDOWBITS`, or `15`.
+         *
+         * @see {@link http://nodejs.org/api/zlib.html#zlib_memory_usage_tuning|Node.js documentation}
          */
         windowBits?: number;
 
         /**
-         * See https://github.com/expressjs/compression#filter regarding the usage.
+         * A function to decide if the response should be considered for compression. This function is called as
+         * `filter(req, res)` and is expected to return `true` to consider the response for compression, or `false` to
+         * not compress the response.
+         *
+         * The default filter function uses the `compressible` module to determine if `res.getHeader('Content-Type')`
+         * is compressible.
+         *
+         * @see {@link https://github.com/expressjs/compression#filter|`filter` documentation}
+         * @see {@link https://www.npmjs.com/package/compressible|compressible module}
          */
         filter?: CompressionFilter;
-
-        /**
-         * See https://nodejs.org/api/zlib.html#zlib_class_options regarding the usage.
-         */
-        flush?: number;
```

Comment:
Why'd you remove these two? They're not explicitly called out in `compression`'s readme, but they are passed thru and used by the underlying `zlib` instance.

Reply:
Thanks for pointing this out. Reverted deletion in PR #37117 since this is now an orphan PR